src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class MethodBuilderImpl : MethodBuilder
+    {
+        internal Type? _returnType;
```

Comment:
Return the `_returnType` from the `ReturnType` property? Similar for the other fields.